File: extract_pdf.js
const fs = require("fs");
let pdfFunc;
try {
  pdfFunc = require("pdf-parse/lib/pdf-parse.js");
} catch (e) {
  const pdf = require("pdf-parse");
  pdfFunc = pdf.default || pdf;
}
const pdfPath =
  "c:\\Users\\reeva.satija_oncehub\\Downloads\\Introduction to DevCenter.pdf";
const outPath = "build/extracted_introduction.txt";

const dataBuffer = fs.readFileSync(pdfPath);

pdfFunc(dataBuffer)
  .then(function (data) {
    fs.mkdirSync("build", { recursive: true });
    fs.writeFileSync(outPath, data.text, "utf8");
    console.log("WROTE: " + outPath);
  })
  .catch(function (err) {
    console.error(err);
    process.exit(1);
  });

File: extract_pdf.py
from pypdf import PdfReader
import sys
from pathlib import Path
pdf_path = r"c:\Users\reeva.satija_oncehub\Downloads\Introduction to DevCenter.pdf"
out_path = Path("build/extracted_introduction.txt")
reader = PdfReader(pdf_path)
texts = []
for p in reader.pages:
    t = p.extract_text()
    if t:
        texts.append(t)
out_path.parent.mkdir(parents=True, exist_ok=True)
out_path.write_text("\n\n".join(texts), encoding="utf-8")
print(f"WROTE: {out_path}")
